tests/rust/pipe-test/async_close/mod.rs

```diff
@@ -131,7 +131,9 @@ fn async_close_pipe_multiple_times_2(libos: &mut LibOS, pipe_name: &str) -> Resu
     // Poll once to ensure the async_close() coroutine runs and finishes the close.
     match libos.wait(qt1, Some(Duration::from_micros(0))) {
         Ok(qr) if qr.qr_opcode == demi_opcode_t::DEMI_OPC_CLOSE && qr.qr_ret == 0 => closed = true,
-        Ok(qr) if qr.qr_opcode == demi_opcode_t::DEMI_OPC_FAILED && qr.qr_ret == libc::ECANCELED => cancelled = true,
+        Ok(qr) if qr.qr_opcode == demi_opcode_t::DEMI_OPC_FAILED && qr.qr_ret == libc::ECANCELED as i64 => {
```

Comment:
Does it make sense to cast the libc errno as i64 or would it make more sense to cast qr.qr_ret as i32?

Reply:
No it does not make sense. We should always cast from the smallest to the largest number-space when comparing. Example:

```
let x: u64 = 0xffffffff_ffffffff;
let y: u32 = 0xffffffff;
(x as u32) == y // This comparison wrongly suceeds.
x == (y as u64) // This comparison correctly fails.
```